app/scripts/metamask-controller.js

```diff
@@ -3703,6 +3704,17 @@ export default class MetamaskController extends EventEmitter {
     return address;
   }
 
+  /**
+   * Get the addresses of the accounts managed by a given Snap.
+   *
+   * @param {string} snapId - Snap ID to get accounts for.
+   * @returns {Promise<string[]>} The addresses of the accounts.
+   */
+  async getAccountsBySnapId(snapId) {
+    const snapKeyring = await this.getSnapKeyring();
+    return await snapKeyring.getAccountsBySnapId(snapId);
+  }
+
```

Comment:
A suggestion for us to consider moving forward, this file is getting too bloated, we should do a refactor to extract some of our logic to another file

Reply:
can this be added to the builder inside [snap-keyring.ts](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/snap-keyring/snap-keyring.ts)?